packages/tables/frontend/sirius-components-tables/src/table/TableContent.tsx

```diff
@@ -32,38 +32,91 @@ const useStyles = makeStyles()((theme) => ({
   },
 }));
 
-export const TableContent = memo(({ editingContextId, representationId, table, readOnly }: TableProps) => {
-  const { classes } = useStyles();
-  const { selection, setSelection } = useSelection();
+export const TableContent = memo(
+  ({ editingContextId, representationId, table, readOnly, onPaginationChange }: TableProps) => {
+    const { classes } = useStyles();
+    const { selection, setSelection } = useSelection();
 
-  const { columns } = useTableColumns(editingContextId, representationId, table, readOnly);
+    const { columns } = useTableColumns(editingContextId, representationId, table, readOnly);
 
-  const muiTable = useMaterialReactTable({
-    columns,
-    data: table.lines,
-    editDisplayMode: 'cell',
-    enableEditing: !readOnly,
-    enableStickyHeader: true,
-    muiTableBodyRowProps: ({ row }) => {
-      return {
-        onClick: () => {
-          const newSelection: Selection = { entries: [{ id: row.original.targetObjectId, kind: 'Object' }] };
-          setSelection(newSelection);
-        },
-        selected: selection.entries.map((entry) => entry.id).includes(row.original.targetObjectId),
-        className: classes.rowMain,
-        sx: {
-          backgroundColor: 'transparent', // required to remove the default mui backgroundColor that is defined as !important
-          cursor: 'pointer',
-        },
-      };
-    },
-    renderTopToolbarCustomActions: () => (
-      <Box>
-        <ExportAllDataButton table={table} />
-      </Box>
-    ),
-  });
+    const [pagination, setPagination] = useState<TablePaginationState>({
+      pageIndex: 0,
+      size: 10,
+      cursor: null,
+      direction: null,
+    });
+
+    const [mRTPagination, setMRTPagination] = useState<MRT_PaginationState>({
+      pageIndex: 0,
+      pageSize: 10,
+    });
```

Comment:
We are explicitly not using `pageIndex` because we do not rely on pages. We should not use `MRT_PaginationState` and Material React Table native pagination support.